contracts/utils/contracts/src/v08/LibBytesV08.sol

```diff
@@ -0,0 +1,440 @@
+            );
+        }
+
+        // Arrays are prefixed by a 256 bit length parameter
+        index += 32;
+
+        // Read the bytes32 from array memory
+        assembly {
+            result := mload(add(b, index))
+        }
+        return result;
+    }
+
+    /// @dev Writes a bytes32 into a specific position in a byte array.
+    /// @param b Byte array to insert <input> into.
+    /// @param index Index in byte array of <input>.
+    /// @param input bytes32 to put into byte array.
+    function writeBytes32(bytes memory b, uint256 index, bytes32 input) internal pure {
+        if (b.length < index + 32) {
+            LibRichErrorsV08.rrevert(
+                LibBytesRichErrorsV08.InvalidByteOperationError(
+                    LibBytesRichErrorsV08.InvalidByteOperationErrorCodes.LengthGreaterThanOrEqualsThirtyTwoRequired,
+                    b.length,
+                    index + 32
+                )
+            );
+        }
+
+        // Arrays are prefixed by a 256 bit length parameter
+        index += 32;
+
+        // Read the bytes32 from array memory
+        assembly {
+            mstore(add(b, index), input)
+        }
+    }
+
+    /// @dev Reads a uint256 value from a position in a byte array.
+    /// @param b Byte array containing a uint256 value.
+    /// @param index Index in byte array of uint256 value.
+    /// @return result uint256 value from byte array.
+    function readUint256(bytes memory b, uint256 index) internal pure returns (uint256 result) {
+        result = uint256(readBytes32(b, index));
+        return result;
+    }
+
+    /// @dev Writes a uint256 into a specific position in a byte array.
+    /// @param b Byte array to insert <input> into.
+    /// @param index Index in byte array of <input>.
+    /// @param input uint256 to put into byte array.
+    function writeUint256(bytes memory b, uint256 index, uint256 input) internal pure {
+        writeBytes32(b, index, bytes32(input));
+    }
+
+    /// @dev Reads an unpadded bytes4 value from a position in a byte array.
+    /// @param b Byte array containing a bytes4 value.
+    /// @param index Index in byte array of bytes4 value.
+    /// @return result bytes4 value from byte array.
+    function readBytes4(bytes memory b, uint256 index) internal pure returns (bytes4 result) {
+        if (b.length < index + 4) {
+            LibRichErrorsV08.rrevert(
+                LibBytesRichErrorsV08.InvalidByteOperationError(
+                    LibBytesRichErrorsV08.InvalidByteOperationErrorCodes.LengthGreaterThanOrEqualsFourRequired,
+                    b.length,
+                    index + 4
+                )
+            );
+        }
+
+        // Arrays are prefixed by a 32 byte length field
+        index += 32;
+
+        // Read the bytes4 from array memory
+        assembly {
+            result := mload(add(b, index))
+            // Solidity does not require us to clean the trailing bytes.
+            // We do it anyway
```

Comment:
Uhh... ok. Why do this if it's just wasting gas and you can't rely on those bits being clear later?